internal/super/generate.go

```diff
@@ -0,0 +1,192 @@
+		Example: "flow generate contract HelloWorld",
+		Args:    cobra.ExactArgs(1),
+	},
+	Flags: &generateFlags,
+	RunS:  generateContract,
+}
+
+var GenerateTransactionCommand = &command.Command{
+	Cmd: &cobra.Command{
+		Use:     "transaction <name>",
+		Short:   "Generate a new transaction",
+		Example: "flow generate transaction SomeTransaction",
+		Args:    cobra.ExactArgs(1),
+	},
+	Flags: &generateFlags,
+	RunS:  generateTransaction,
+}
+
+var GenerateScriptCommand = &command.Command{
+	Cmd: &cobra.Command{
+		Use:     "script <name>",
+		Short:   "Generate a new script",
+		Example: "flow generate script SomeScript",
+		Args:    cobra.ExactArgs(1),
+	},
+	Flags: &generateFlags,
+	RunS:  generateScript,
+}
+
+func init() {
+	GenerateContractCommand.AddToParent(GenerateCommand)
+	GenerateTransactionCommand.AddToParent(GenerateCommand)
+	GenerateScriptCommand.AddToParent(GenerateCommand)
+}
+
+func generateContract(
+	args []string,
+	_ command.GlobalFlags,
+	logger output.Logger,
+	flow flowkit.Services,
+	state *flowkit.State,
+) (result command.Result, err error) {
+	return generateNew(args, "contract", logger)
+}
+
+func generateTransaction(
+	args []string,
+	_ command.GlobalFlags,
+	logger output.Logger,
+	flow flowkit.Services,
+	state *flowkit.State,
+) (result command.Result, err error) {
+	return generateNew(args, "transaction", logger)
+}
+
+func generateScript(
+	args []string,
+	_ command.GlobalFlags,
+	logger output.Logger,
+	flow flowkit.Services,
+	state *flowkit.State,
+) (result command.Result, err error) {
+	return generateNew(args, "script", logger)
+}
+
+func generateNew(
+	args []string,
+	templateType string,
+	logger output.Logger,
+) (result command.Result, err error) {
+	if len(args) < 1 {
+		return nil, fmt.Errorf("invalid number of arguments")
+	}
+
+	name := args[0]
+	var filename string
+
+	// Don't add .cdc extension if it's already there
+	if strings.HasSuffix(name, ".cdc") {
+		filename = name
+	} else {
+		filename = fmt.Sprintf("%s.cdc", name)
+	}
+
+	var fileToWrite string
+	var basePath string
+
+	if generateFlags.Directory != "" {
+		basePath = generateFlags.Directory
+	} else {
+		switch templateType {
+		case "contract":
+			basePath = "cadence/contracts"
+		case "script":
+			basePath = "cadence/scripts"
+		case "transaction":
+			basePath = "cadence/transactions"
+		default:
+			return nil, fmt.Errorf("invalid template type: %s", templateType)
+		}
+	}
+
+	switch templateType {
+	case "contract":
+		fileToWrite = fmt.Sprintf(`
+access(all) contract %s {
+    init() {}
+}`, name)
+	case "script":
+		fileToWrite = `access(all) fun main() {
+    // Script details here
+}`
+	case "transaction":
+		fileToWrite = `transaction() {
+    prepare(account:AuthAccount) {}
+
+    execute {}
+}`
+	default:
+		return nil, fmt.Errorf("invalid template type: %s", templateType)
+	}
+
+	filenameWithBasePath := filepath.Join(basePath, filename)
+
+	if _, err := os.Stat(filenameWithBasePath); err == nil {
```

Comment:
flow-cli has the `--save` `-s` flag to save the output of a command to a file, do we need to have this logic here or can we just return the output and rely on that logic?

also state has a ReaderWriter that is used to save files and that is used elsewhere instead of os functions.

Reply:
Hmm yeah I guess that means that you always have to provide `--save`. I was hoping to just always generate them to the correct folders unless specified otherwise. But it makes sense to keep the same pattern. Just really verbose now.

Reply:
Imo just skip the pattern/flag here.  I think think it's pretty obvious that this will be saving a file and using a flag for this is just redundant & a command like this is for ease of use so should be as shorthand as possible.  I think the only purpose `--save` potentially has in this is as an override to the output path.

Reply:
I'm fully in support for super short aliases here as well (i.e. `flow g c FooBar`)

Reply:
Feel free to not use --save. But using ReaderWriter is recommended. I think it also creates required directories.

Reply:
Yeah ReaderWriter is no problem. Good catch. Will change that @bjartek